pandas/core/style.py

```diff
@@ -0,0 +1,511 @@
+    def _color_bg_range(s, cmap='PuBu', low=0, high=0):
+        """Color background in a range according to the data."""
+        with mpl(Styler.background_gradient.__name__) as (plt, colors):
+            norm = colors.Normalize(s.min() * (1 + low), s.max() * (1 + high))
+            normed = norm(s.values)  # matplotlib modifies?
+            c = [colors.rgb2hex(x) for x in plt.cm.get_cmap(cmap)(normed)]
+            return ['background-color: %s' % color for color in c]
+
+    def text_shadow(self, color='black'):
+        self.set_properties(**{"text-shadow": "-1px 0 {color}, 0 1px {color}, "
+                               "1px 0 {color}, 0 -1px {color};".format(color=color)})
+        return self
+
+    def set_properties(self, subset=None, **kwargs):
+        """
+        Convience method for setting a non-data dependent properties
+        on each element.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        kwargs: dict
+            property: value pairs to be set for each cell
+        subset: IndexSlice
+            a valid slice for ``data``
+
+        Returns
+        -------
+        self : Styler
+
+        Examples
+        --------
+        df.stle.set_properties(color="white", align="right")
+        """
+        values = ';'.join('{p}: {v}'.format(p=p, v=v) for p, v in kwargs.items())
+        f = lambda x: values
+        if self.data.ndim == 2:
+            return self.applymap(f, subset=subset)
+        else:
+            return self.apply(f, subset=subset)
+
+    @staticmethod
+    def _bar(s, color, width):
+        normed = width * (s - s.min()) / (s.max() - s.min())
+        attrs = 'width: 10em; height: 80%;'\
+                'background: linear-gradient(90deg,'\
+                                            '{c} {w}%, transparent 0%)'
+        return [attrs.format(c=color, w=x) for x in normed]
+
+    def bar(self, subset=None, axis=0, color='#FFC0CB', width=100):
+        """
+        Color the background `color` proptional to the values in each column.
+        Excludes non-numeric data by default.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        subset: IndexSlice, default None
+        axis: int
+        color: str
+        width: float
+            A number between 0 or 100. The largest value will cover this
+            percent of the cell's width
+        """
+        subset = _maybe_numeric_slice(self.data, subset)
+        subset = self._non_reducing_slice(subset)
+        result = self.apply(self._bar, subset=subset, axis=axis, color=color,
+                            width=width)
+        return self
+
+    @staticmethod
+    def _non_reducing_slice(slice_):
+        """
+        Ensurse that a slice doesn't reduce to a Series or Scalar.
+        """
+        # default to column slice, like DataFrame
+        # ['A', 'B'] -> IndexSlices[:, ['A', 'B']]
+        if isinstance(slice_, (pd.Series, np.ndarray, pd.Index, list)):
```

Comment:
Looks like we can't easily support tuples here `df.style.apply(func, subset=('A', 'B')]` since then it's ambiguous whether you're slicing the row 'A', and column 'B' or the columns 'A' and 'B' (or the column with the key `("A", "B")`). We don't have that same problem with lists.